### PR TITLE
feat(test-infra): Bump to latest version

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
- Bump test-infra definition to the latest version to be able to run `new-e2e` tests running on Agent6
- Set `6.53.x` as `main` on this branch ([ACIX-467](https://datadoghq.atlassian.net/browse/ACIX-467))

### Motivation
https://datadoghq.atlassian.net/browse/ACIX-496

### Describe how to test/QA your changes
tests/linters are passing on the branch

### Possible Drawbacks / Trade-offs

Pipeline status:
- tests_deb-arm64-py3: [fails](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/703992001) due to `panic / race condition` This is maybe an incompatibility due to something not fixed related to the test-infra bump. I would like to accept this failure and review it when we start fixing the tests.
- kmt_testing: must restore the images used for the tests
- e2e: I would like to start fixing-up in dedicated PR
- kitchen: https://github.com/DataDog/datadog-agent/pull/31025

Besides, I've launched a full pipeline on this PR to assess the status:
- qa_agent: issue with windows image name (`severcore`, because `nanoserver` [does not support python2](https://github.com/DataDog/datadog-agent/blob/7.53.x/.gitlab/container_build/docker_windows_agent6.yml#L2))
- install_scripts: they still run with agent7

[ACIX-467]: https://datadoghq.atlassian.net/browse/ACIX-467?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by chouetz** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: junit-local.tgz
Comment (chouetz) [inline junit-local.tgz:1]: oops, removed on the last commit

**Review by CelianR — APPROVED** [review]
Thanks for the commit-by-commit review! LGTM

File: .gitlab/internal_image_deploy/internal_image_deploy.yml
Comment (AliDatadog) [inline .gitlab/internal_image_deploy/internal_image_deploy.yml:1]: I think we don't need this job. This is for internal deployments only

**Review by Jakobovski — APPROVED** [review]

**Review by sabrina-datadog — APPROVED** [review]

**Review by louis-cqrl — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**